File: Sparkmax/SparkmaxBasicSetJava/src/main/java/frc/robot/Robot.java
package frc.robot;

import edu.wpi.first.wpilibj.TimedRobot;
//0. import PWMSparkMax (VS code will automatically do this after you declare your motor)
import edu.wpi.first.wpilibj.motorcontrol.PWMSparkMax;

public class Robot extends TimedRobot {
  //1. declare PWMSpark max motor
  public PWMSparkMax basicMotor;

  @Override
  public void robotInit() {
    //2. Initialize basic motor to use PWM channel Zero
    basicMotor = new PWMSparkMax(0);
  }

  @Override
  public void teleopPeriodic() {
    //3. Run the motor at 50% speed counter clockwise
    basicMotor.set(0.5);
  }
}

File: Sparkmax/SparkmaxBasicSetCpp/src/main/cpp/Robot.cpp
#include "Robot.h"

#include <fmt/core.h>
//1. import PWMSparkMax library
#include "frc/motorcontrol/PWMSparkMax.h"

//2. declare basic motor object with PWM port number inbetween {}
frc::PWMSparkMax basicMotor{0};

void Robot::TeleopPeriodic() {
  //3. set motor to run 50% speed counter clockwise
  basicMotor.Set(0.5);
}

void Robot::RobotInit() {
}

void Robot::RobotPeriodic() {}

void Robot::AutonomousInit() {
}

void Robot::AutonomousPeriodic() {
}

void Robot::TeleopInit() {}

void Robot::DisabledInit() {}

void Robot::DisabledPeriodic() {}

void Robot::TestInit() {}

void Robot::TestPeriodic() {}

void Robot::SimulationInit() {}

void Robot::SimulationPeriodic() {}

#ifndef RUNNING_FRC_TESTS
int main() {
  return frc::StartRobot<Robot>();
}
#endif
